pkg/services/networkverifier/networkverifier.go

```diff
@@ -0,0 +1,221 @@
+		}
+		c.cd = cd
+	}
+	// fmt.Printf("cd ::: %v\n", cd)
+	return nil
+}
+
+type egressConfig struct {
+	cloudImageID string
+	instanceType string
+	cloudTags    map[string]string
+	timeout      time.Duration
+	kmsKeyID     string
+	httpProxy    string
+	httpsProxy   string
+	CaCert       string
+	noTLS        bool
+}
+
+var (
+	awsDefaultTags = map[string]string{"osd-network-verifier": "owned", "red-hat-managed": "true", "Name": "osd-network-verifier"}
+)
+
+// VerifierResult type
+type VerifierResult int
+
+// Verifier outcomes
+const (
+	Undefined VerifierResult = 0
+	Failure
+	Success
+)
+
+// RunNetworkVerifier runs the network verifier tool to check for network misconfigurations
+func (c Client) RunNetworkVerifier(externalClusterID string) (VerifierResult, string, error) {
+	fmt.Printf("Running Network Verifier...\n")
+	err := c.populateStructWith(externalClusterID)
+	if err != nil {
+		return Undefined, "", fmt.Errorf("failed to populate struct in runNetworkVerifier in networkverifier step: %w", err)
+	}
+
+	infraID := c.cd.Spec.ClusterMetadata.InfraID
+
+	credentials, err := c.GetAWSCredentials()
+	if err != nil {
+		return Undefined, "", fmt.Errorf("failed to get AWS Credentials: %w", err)
+	}
+
+	config := egressConfig{}
+
+	p := proxy.ProxyConfig{
+		HttpProxy:  config.httpProxy,
+		HttpsProxy: config.httpsProxy,
+		Cacert:     config.CaCert,
+		NoTls:      config.noTLS,
+	}
+
+	securityGroupID, err := c.GetSecurityGroupID(infraID)
+	if err != nil {
+		return Undefined, "", fmt.Errorf("failed to get SecurityGroupId: %w", err)
+	}
+
+	subnets, err := c.GetSubnets(infraID)
+	subnet := subnets[0]
+	if err != nil {
+		return Undefined, "", fmt.Errorf("failed to get Subnets: %w", err)
+	}
+
+	fmt.Printf("Using Security Group ID: %s\n", securityGroupID)
+	fmt.Printf("Using SubnetID: %s\n", subnet)
+
+	// setup non cloud config options
+	vei := verifier.ValidateEgressInput{
+		Ctx:          context.TODO(),
+		SubnetID:     subnet,
+		CloudImageID: config.cloudImageID,
+		Timeout:      config.timeout,
+		Tags:         config.cloudTags,
+		InstanceType: config.instanceType,
+		Proxy:        p,
+	}
+
+	if len(vei.Tags) == 0 {
+		vei.Tags = awsDefaultTags
+	}
+
+	// Setup AWS Specific Configs
+	vei.AWS = verifier.AwsEgressConfig{
+		KmsKeyID:        config.kmsKeyID,
+		SecurityGroupId: securityGroupID,
+	}
+
+	awsVerifier, err := awsverifier.NewAwsVerifier(credentials.AccessKeyID, credentials.SecretAccessKey, credentials.SessionToken, c.cluster.Region().ID(), "", true)
+	if err != nil {
+		return Undefined, "", fmt.Errorf("could not build awsVerifier %v", err)
+	}
+
+	awsVerifier.Logger.Warn(context.TODO(), "Using region: %s", c.cluster.Region().ID())
+
+	out := verifier.ValidateEgress(awsVerifier, vei)
+
+	verifierFailures, verifierExceptions, verifierErrors := out.Parse()
+
+	if len(verifierExceptions) != 0 && len(verifierErrors) != 0 {
+		exceptionsSummary := verifierExceptions[0].Error()
+		errorsSummary := verifierErrors[0].Error()
```

Comment:
**Question:** Can we safely ignore the other values in the `exceptionsSummary`, `errorsSummary` arrays? We might want to pick them up as well.

Reply:
This picks up everything.